### DISCUSSION TITLE
[RFC] Workflow - Request Review

### DISCUSSION THREAD

**Discussion by leevigraham** [craftcms/cms#8828] [Ideas]
### Description

A summary of discussions with @brandonkelly about new workflow / permissions functionality.

Given the current permissions an author can create drafts but not publish entries. To have the entry published they need to manually contact a user with publish permissions.

In the case where you have a lot of users who can register themselves and enter content they may not know who / how to contact.

I propose that authors should be able to trigger a review request from the entry edit screen (or event the entry control panel index).

This could work by assigning a new permission to users: "Review Drafts" under "Publish Entries" which would identify a user or group as a reviewer for the element.

![image](https://user-images.githubusercontent.com/25124/81032443-1f254000-8ed3-11ea-869e-6b1682d9c48c.png)

An author could request a review by just clicking a button (all reviewers would be notified) or autocomplete select one or more users.

This functionality could mirror githubs review process:

* https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews
* https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/requesting-a-pull-request-review

**Comment by green17** [comment]
This would be great to be baked, the ability for an editor to:

1. Submit for review
2. Content is locked to all users until reviewed
3. User gets an email when approved
4. Option for editors to duplicate an entry as a draft

**Comment by acekevin** [comment]
Another flow option along with Github is what Contentful does with Tasks. It keeps the idea of approval pretty agnostic to whatever the team wants to make a "task" be.

https://www.contentful.com/help/tasks/

**Reply by leevigraham** [reply]
Hmmm… Tasks could be quite interesting.

**Reply by brandonkelly** [reply]
Yeah that’s closer to what we are thinking ;)
